lnrpc/peersrpc/peers_server.go

```diff
@@ -0,0 +1,142 @@
+	PeersServer
+}
+
+// Server is a sub-server of the main RPC server: the peers RPC. This sub
+// RPC server allows to intereact with our Peers in the Lightning Network.
+type Server struct {
+	started  int32 // To be used atomically.
+	shutdown int32 // To be used atomically.
+
+	// Required by the grpc-gateway/v2 library for forward compatibility.
+	// Must be after the atomically used variables to not break struct
+	// alignment.
+	UnimplementedPeersServer
+
+	cfg *Config
+}
+
+// A compile time check to ensure that Server fully implements the PeersServer
+// gRPC service.
+var _ PeersServer = (*Server)(nil)
+
+// New returns a new instance of the peersrpc Peers sub-server. We also
+// return the set of permissions for the macaroons that we may create within
+// this method. If the macaroons we need aren't found in the filepath, then
+// we'll create them on start up. If we're unable to locate, or create the
+// macaroons we need, then we'll return with an error.
+func New(cfg *Config) (*Server, lnrpc.MacaroonPerms, error) {
+	server := &Server{
+		cfg: cfg,
+	}
+
+	return server, macPermissions, nil
+}
+
+// Start launches any helper goroutines required for the Server to function.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Start() error {
+	if atomic.AddInt32(&s.started, 1) != 1 {
+		return nil
+	}
+
+	return nil
+}
+
+// Stop signals any active goroutines for a graceful closure.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Stop() error {
+	if atomic.AddInt32(&s.shutdown, 1) != 1 {
+		return nil
+	}
+
+	return nil
+}
+
+// Name returns a unique string representation of the sub-server. This can be
+// used to identify the sub-server and also de-duplicate them.
+//
+// NOTE: This is part of the lnrpc.SubServer interface.
+func (s *Server) Name() string {
+	return subServerName
+}
+
+// RegisterWithRootServer will be called by the root gRPC server to direct a
+// sub RPC server to register itself with the main gRPC root server. Until this
+// is called, each sub-server won't be able to have
+// requests routed towards it.
+//
+// NOTE: This is part of the lnrpc.GrpcHandler interface.
+func (r *ServerShell) RegisterWithRootServer(grpcServer *grpc.Server) error {
+	// We make sure that we register it with the main gRPC server to ensure
+	// all our methods are routed properly.
+	RegisterPeersServer(grpcServer, r)
+
+	log.Debugf("Peers RPC server successfully register with root " +
+		"gRPC server")
+
+	return nil
+}
+
+// RegisterWithRestServer will be called by the root REST mux to direct a sub
+// RPC server to register itself with the main REST mux server. Until this is
+// called, each sub-server won't be able to have requests routed towards it.
+//
+// NOTE: This is part of the lnrpc.GrpcHandler interface.
+func (r *ServerShell) RegisterWithRestServer(ctx context.Context,
+	mux *runtime.ServeMux, dest string, opts []grpc.DialOption) error {
+
+	return nil
```

Comment:
We should enable the REST proxy for this service. You might need to run `make rpc` on this commit to generate the required `peers.pb.gw.go` file.

Reply:
I did not notice this one! We could add a `make rpc` step in our CI and fail if there is any difference with the ones that would be generated/formatted in the last commit :ok_hand:

Reply:
We do have a CI step. But apparently it doesn't work for _new_ files, since `git diff` doesn't report anything for new files that haven't been added.

Reply:
This hasn't been fixed yet. Forgot to push a commit?

Reply:
yep, not sure what happened here, my editor was showing me something different from this, `git push -f` was telling me that everything was already up to date and `git status` was showing nothing new. I made a small change, force pushed and everything seems up to date now